synapse/lib/httpapi.py

```diff
@@ -280,7 +280,19 @@ class AuthUsersV1(Handler):
         if not await self.reqAuthUser():
             return
 
-        self.sendRestRetn([u.pack() for u in self.cell.auth.users()])
+        try:
+
+            archived = int(self.get_argument('archived', default='0'))
+
+        except Exception as e:
+            return self.sendRestErr('BadHttpParam', 'The parameter "archived" must be 0 or 1 if specified.')
```

Comment:
This actually allows values outside of `(0, 1)` to be passed in via API.

Reply:
added a check